modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -493,7 +493,9 @@ void reset(final Status st) {
 
                     if (listener != null) {
                         if (status.isOk()) {
-                            listener.onNewPeersConfigurationApplied(resultPeerIds, resultLearnerIds);
+                            LogId id = node.conf.getId();
```

Comment:
Did you check that node.conf.getId() matches currently processing configuration change? Meaning that it's not the old one.

Reply:
discussed in private